domain/secret/grant.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package secret
+
+import coresecrets "github.com/juju/juju/core/secrets"
+
+// Role represents the role of a secret access grant
+// as recorded in the secret_role lookup table.
+type Role int
+
+const (
+	RoleNone Role = iota
+	RoleView
+	RoleManage
+)
+
+// MarshallRole converts a secret role to a db role id.
+func MarshallRole(role coresecrets.SecretRole) Role {
+	switch role {
+	case coresecrets.RoleView:
+		return RoleView
+	case coresecrets.RoleManage:
+		return RoleManage
+	}
+	return RoleNone
+}
+
+// GrantScopeType represents the type of a subject
+// granted access to a secret as recorded in the
+// secret_grant_subject_type lookup table.
+type GrantScopeType int
+
+const (
+	ScopeUnit GrantScopeType = iota
+	ScopeApplication
+	ScopeModel
+	ScopeRelation
+)
+
+// GoString implements fmt.GoStringer.
+func (s GrantScopeType) GoString() string {
+	switch s {
+	case ScopeUnit:
+		return "unit"
```

Comment:
These values look ok for String(), but it'd be better to indicate the Type in GoString() result.

Reply:
Used String()